Added Java version of subclassing

brandon-sensibill opened the pull request.

brandon-sensibill committed:
Added Java version of subclassing

patshu commented on app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassJavaActivity.java:
```diff
@@ -0,0 +1,149 @@
+     */
+    @Override
+    protected int getWebContainerId() {
+        return binding.webContainer.getId();
```
Should we do this or `R.id.webContainer`?  I guess i just never thought about it but my first thought would be to use R

brandon-sensibill replied:
I dont think there is any difference. Going through R. would add another import. I dont see anything online that suggests 1 or or the other

brandon-sensibill replied:
I dont think it matters, but updated both the java and kotlin to just use the R

patshu commented on app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassJavaActivity.java:
```diff
@@ -0,0 +1,149 @@
+    @Override
+    protected @NotNull WebUiFragment createWebUiFragment() {
+        WebUiFragment fragment = super.createWebUiFragment();
+        Bundle bundle = new Bundle();
```
Just to be consistent with the kotlin example I'd suggest
```java
Bundle bundle = fragment.getArguments();
if (bundle == null) {
    bundle = new Bundle();
}
```

brandon-sensibill replied:
Updated

patshu commented on app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassJavaActivity.java:
```diff
@@ -0,0 +1,149 @@
+    @Override
+    public void onDisplayNetworkError(boolean networkNotAvailable) {
+        super.onDisplayNetworkError(networkNotAvailable);
+        Timber.i(networkNotAvailable ? "true" : "false");
```
It's been so long since we haven't had the convenience of Kotlin haha.
But just to check, can we use `String.valueOf(nna)`?

brandon-sensibill replied:
looks like it. updated

patshu commented on app/src/main/java/com/getsensibill/spendmanager/demo/subclass/SubclassJavaActivity.java:
```diff
@@ -0,0 +1,149 @@
+    @Override
+    protected @NotNull WebUiNetworkErrorFragment createWebUiNetworkErrorFragment() {
+        WebUiNetworkErrorFragment fragment = super.createWebUiNetworkErrorFragment();
+        Bundle bundle = new Bundle();
```
Same comment as the arguments for the web fragment

brandon-sensibill replied:
updated

brandon-sensibill committed:
Updated the webContainerId to use the R. instead of view binding to get the container id.
Updated bundle loading for the java version, so it maintans any pre existing bundle properties.
